skyportal/tests/frontend/test_sources.py

```diff
@@ -55,29 +56,70 @@ def test_classifications(driver, user, taxonomy_token, public_group, public_sour
         ],
     }
 
+    tax_name = str(uuid.uuid4())
+    tax_version = "test0.1"
+
     status, data = api(
         'POST',
         'taxonomy',
         data={
-            'name': str(uuid.uuid4()),
+            'name': tax_name,
             'hierarchy': simple,
             'group_ids': [public_group.id],
-            'version': "test0.1",
+            'version': tax_version,
         },
         token=taxonomy_token,
     )
     assert status == 200
 
-    driver.get(f"/become_user/{user.id}")  # TODO decorator/context manager?
+    driver.get(f"/become_user/{user.id}")
     driver.get(f"/source/{public_source.id}")
     driver.wait_for_xpath(f'//div[text()="{public_source.id}"]')
-    taxonomy_button = driver.wait_for_xpath(f'//div[@id="tax-select"]')
-    taxonomy_button.click()
+    driver.wait_for_xpath_to_be_clickable('//div[@id="tax-select"]').click()
+    driver.wait_for_xpath_to_be_clickable(
+        f'//*[text()="{tax_name} ({tax_version})"]'
+    ).click()
+    driver.wait_for_xpath("//body").click()
```

Comment:
why isn't the wait for `body` happening sooner? Like before `'//div[@id="tax-select"]'`?

Reply:
@profjsb this is just to fully close the MUI select.

Reply:
See, e.g., `skyportal/tests/frontend/test_upload_photometry.py`